<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>fermentableEditor</class>
 <widget class="QDialog" name="fermentableEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>370</width>
    <height>367</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Fermentable Editor</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="configSection" stdset="0">
   <string notr="true">fermentableEditor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabFermentablesDetails">
      <attribute name="title">
       <string>Details</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Name</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_name</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEdit_name">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Name</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="BtColorLabel" name="label_color">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>Color</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_color</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="BtColorEdit" name="lineEdit_color">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Lovibond rating</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">color_srm</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="BtMassLabel" name="label_amount">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>Default Amount</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_amount</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="BtMassEdit" name="lineEdit_amount">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Amount</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">amount_kg</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_yield">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Yield (%)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_yield</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="BtGenericEdit" name="lineEdit_yield">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Yield as compared to glucose</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">yield_pct</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Type</string>
         </property>
         <property name="buddy">
          <cstring>comboBox_type</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="comboBox_type">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Type</string>
         </property>
         <item>
          <property name="text">
           <string>Grain</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sugar</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Extract</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Dry Extract</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Adjunct</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_addAfterBoil">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Add after boil</string>
         </property>
         <property name="buddy">
          <cstring>checkBox_addAfterBoil</cstring>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QCheckBox" name="checkBox_addAfterBoil">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>This ingredient is added post boil.</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_recommendMash">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Recommend mash</string>
         </property>
         <property name="buddy">
          <cstring>checkBox_recommendMash</cstring>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="checkBox_recommendMash">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Recommend this be mashed</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_isMashed">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Is mashed</string>
         </property>
         <property name="buddy">
          <cstring>checkBox_isMashed</cstring>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QCheckBox" name="checkBox_isMashed">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Checked if it is present in mash</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabFermentablesProperties">
      <attribute name="title">
       <string>Properties</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_coarseFineDiff">
         <property name="text">
          <string>Coarse/Fine Diff (%)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_coarseFineDiff</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="BtGenericEdit" name="lineEdit_coarseFineDiff">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Yield difference between coarse and fine grind</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">coarseFineDiff_pct</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_moisture">
         <property name="text">
          <string>Moisture (%)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_moisture</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="BtGenericEdit" name="lineEdit_moisture">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Moisture percentage by mass</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">moisture_pct</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_diastaticPower">
         <property name="text">
          <string>DP (Lintner)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_diastaticPower</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="BtGenericEdit" name="lineEdit_diastaticPower">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Diastatic power in Lintner</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">diastaticPower_lintner</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_protein">
         <property name="text">
          <string>Protein (%)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_protein</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="BtGenericEdit" name="lineEdit_protein">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Protein percentage by mass</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">protein_pct</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_maxInBatch">
         <property name="text">
          <string>Max In Batch (%)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_maxInBatch</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="BtGenericEdit" name="lineEdit_maxInBatch">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Maximum recommended percentage of total grist</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">maxInBatch_pct</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_ibuGalPerLb">
         <property name="text">
          <string>Bitterness (IBU*gal/lb)</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_ibuGalPerLb</cstring>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="BtGenericEdit" name="lineEdit_ibuGalPerLb">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Bitterness of pre-hopped extracts</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">ibuGalPerLb</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Info</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_origin">
           <property name="text">
            <string>Origin</string>
           </property>
           <property name="buddy">
            <cstring>lineEdit_origin</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_origin">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Origin</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_supplier">
           <property name="text">
            <string>Supplier</string>
           </property>
           <property name="buddy">
            <cstring>lineEdit_supplier</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEdit_supplier">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Supplier</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="BtMassLabel" name="label_inventory">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="text">
            <string>Amount in inventory</string>
           </property>
           <property name="buddy">
            <cstring>lineEdit_inventory</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="BtMassEdit" name="lineEdit_inventory">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Amount in inventory</string>
           </property>
           <property name="editField" stdset="0">
            <string notr="true">inventory</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_notes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="text">
          <string>Notes:</string>
         </property>
         <property name="buddy">
          <cstring>textEdit_notes</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="textEdit_notes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BtGenericEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>BtColorLabel</class>
   <extends>QLabel</extends>
   <header>BtLabel.h</header>
   <slots>
    <signal>labelChanged(Unit::unitDisplay,Unit::unitScale)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtColorEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
   <slots>
    <slot>lineChanged(Unit::unitDisplay,Unit::unitScale)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtMassLabel</class>
   <extends>QLabel</extends>
   <header>BtLabel.h</header>
   <slots>
    <signal>labelChanged(Unit::unitDisplay,Unit::unitScale)</signal>
    <slot>popContextMenu(QPoint)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtMassEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
   <slots>
    <signal>textModified()</signal>
    <slot>lineChanged()</slot>
    <slot>lineChanged(Unit::unitDisplay,Unit::unitScale)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lineEdit_name</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>fermentableEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>487</x>
     <y>345</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>fermentableEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>345</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>label_amount</sender>
   <signal>labelChanged(Unit::unitDisplay,Unit::unitScale)</signal>
   <receiver>lineEdit_amount</receiver>
   <slot>lineChanged(Unit::unitDisplay,Unit::unitScale)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>label_inventory</sender>
   <signal>labelChanged(Unit::unitDisplay,Unit::unitScale)</signal>
   <receiver>lineEdit_inventory</receiver>
   <slot>lineChanged(Unit::unitDisplay,Unit::unitScale)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>177</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>label_color</sender>
   <signal>labelChanged(Unit::unitDisplay,Unit::unitScale)</signal>
   <receiver>lineEdit_color</receiver>
   <slot>lineChanged(Unit::unitDisplay,Unit::unitScale)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
